### PR TITLE
Remove benthos in v2

### PR DESCRIPTION
**Pull request by ensary**
The primary goal is to get rid of multiple dependencies that benthos is pulling in. 
None of our known services currently use benthos producer, but we keep getting multiple security vulnerability tickets for benthos dependencies.
NB - the PR is for the v2 branch, so any consumers who want to use this new version need to use `github.com/asecurityteam/component-producer/v2` (see https://golang.org/doc/modules/major-version for details). For any existing consumers linking to the library this change is **not** visible unless they explicitly opt in.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,13 +22,15 @@
 This is a [`settings`](https://github.com/asecurityteam/settings) that enables
 constructing a an event producer. The resulting client may be either an HTTP
```
Comment (faldridge) [inline README.md:23]: nit: the wording here should probably be tweaked away from "either" to "Currently, the only supported client is HTTP, as provided by ...".

File: go.sum [hunk 2]
```diff
@@ -1686,290 +610,64 @@ golang.org/x/tools v0.0.0-20181207183836-8bc39b988060/go.mod h1:n7NCudcB/nEzxVGm
...
-honnef.co/go/tools v0.0.1-2020.1.5/go.mod h1:X/FiERA/W4tHapMX5mGpAtMSVEeEUOyHaw9vFzvIQ3k=
-mvdan.cc/gofumpt v0.0.0-20200709182408-4fd085cb6d5f/go.mod h1:9VQ397fNXEnF84t90W4r4TRCQK+pg9f8ugVfyj+S26w=
-mvdan.cc/interfacer v0.0.0-20180901003855-c20040233aed/go.mod h1:Xkxe497xwlCKkIaQYRfC7CSLworTXY9RMqwhhCm+8Nc=
-mvdan.cc/lint v0.0.0-20170908181259-adc824a0674b/go.mod h1:2odslEg/xrtNQqCYg2/jCoyKnw3vv5biOc3JnIcYfL4=
-mvdan.cc/unparam v0.0.0-20190720180237-d51796306d8f/go.mod h1:4G1h5nDURzA3bwVMZIVpwbkw+04kSxk3rAtzlimaUJw=
-rsc.io/binaryregexp v0.2.0/go.mod h1:qTv7/COck+e2FymRvadv62gMdZztPaShugOCi3I+8D8=
 rsc.io/quote/v3 v3.1.0/go.mod h1:yEA65RcK8LyAZtP9Kv3t0HmxON59tX3rD+tICJqUlj0=
 rsc.io/sampler v1.3.0/go.mod h1:T1hPZKmBbMNahiBKFy5HrXp6adAjACjK9JXDnKaTXpA=
-sigs.k8s.io/yaml v1.1.0/go.mod h1:UJmg0vDUVViEyp3mgSv9WPwZCDxu4rQW1olrI1uml+o=
-sourcegraph.com/sourcegraph/appdash v0.0.0-20190731080439-ebfcffb1b5c0/go.mod h1:hI742Nqp5OhwiqlzhgfbWU4mW4yO10fP+LoT9WOswdU=
```
Comment (faldridge) [inline go.sum:1973]: ❤️

File: producer.go [hunk 3]
```diff
@@ -19,10 +17,9 @@ const (
 
 // Config wraps all producer related configuration.
 type Config struct {
-	Type    string `default:"The type of producer. The choices are BENTHOS and POST."`
-	Benthos *BenthosConfig
-	POST    *POSTConfig
-	Null    *NullConfig
+	Type string `default:"The type of producer. The choices are BENTHOS and POST."`
```
Comment (faldridge) [inline producer.go:20]: nit: the choices now are POST and NULL, right?

**Review by faldridge — APPROVED** [review]
Some minor nits on the wording, but this is great, thanks!

File: producer.go [hunk 4]
```diff
@@ -32,35 +29,30 @@ func (*Config) Name() string {
...
 }
 
 // Settings generates a default configuration.
 func (c *Component) Settings() *Config {
 	return &Config{
-		Type:    "BENTHOS",
-		Benthos: c.Benthos.Settings(),
-		POST:    c.POST.Settings(),
-		Null:    c.Null.Settings(),
+		Type: "BENTHOS",
```
Comment (alecbert-atl) [inline producer.go:47]: Following on with the comment above, I imagine our default Type should also be POST or Null

Comment (nyanshak) [inline producer.go:47]: question: Should there still be `BENTHOS` type settings components? I'm not super familiar with this repo.

Comment (ensary) [reply producer.go]: Already fixed :)

**Review by nyanshak — APPROVED** [review]

**Review by faldridge — APPROVED** [review]

**Pull request merged**